MontyHallTests/ViewTests/ViewTests.cs

```diff
@@ -0,0 +1,158 @@
+using System;
+using MontyHallKata.Models.Entity;
+using MontyHallKata.Views;
+using MontyHallKata.Views.InputOutput;
+using Moq;
+using Xunit;
+
+namespace MontyHallTests.ViewTests
+{
+    public class ViewTests
+    {
+        private readonly Mock<IInputOutput> _mockedConsole;
+        private readonly View _view;
+        public ViewTests()
+        {
+            _mockedConsole = new Mock<IInputOutput>();
+            _view = new View(_mockedConsole.Object);
+        }
+        
+        [Theory]
+        [InlineData(GameStatus.Won, InputOutputMessages.WinningOutputMessage)]
+        [InlineData(GameStatus.Lost, InputOutputMessages.LosingOutputMessage)]
+        [InlineData(GameStatus.Quit, InputOutputMessages.QuitOutputMessage)]
+        public void GivenTheGameStatusIsPassedIn_WhenTheValueChanges_ThenTheOutputMessageShouldChangeAccordingly(GameStatus gameStatus, string expectedOutputMessage)
+        {
+            // Arrange 
+            _mockedConsole.Setup(console => console.PrintOutput(expectedOutputMessage));
+            // Act 
+            _view.HandleOutputMessage(gameStatus);
+            
+            // Assert
+            _mockedConsole.Verify();
+        }
+
+        [Fact]
+        public void GivenTheMontyHallView_WhenGetUserChoiceIsCalled_TheUserShouldBePromptedWithTheChoiceToStaySwitchOrQuitAndTheChoiceShouldBeReturned()
+        {
+            // Arrange
+            const int expectedChoice = 1;
+            _mockedConsole.Setup(console => console.GetIntInput())
+                .Returns(expectedChoice);
+            _mockedConsole.Setup(console => console.PrintOutput(InputOutputMessages.ChoicePromptMessage))
+                .Verifiable();
+            
+            // Act 
+            var userChoice = _view.GetUserChoice();
+            
+            // Assert
+            _mockedConsole.Verify();
+            Assert.Equal(expectedChoice, userChoice);
+        }
+
+        [Fact]
+        public void
+            GivenAnInvalidFinalGameStatusIsPassedIn_WhenTheHandleOutputMessageIsCalled_ThenShouldThrowArgumentOutOfRangeException()
+        {
+            // Arrange 
+            var view = new View(new InputOutputConsole());
+            const GameStatus invalidFinalGameStatus = GameStatus.Playing;
+            
+            // Act & Assert 
+            Assert.Throws<ArgumentOutOfRangeException>(() => view.HandleOutputMessage(invalidFinalGameStatus));
```

Comment:
`Assert.Throws` only verifies the type of exception, so can have another assertion to verify error message.

ref: https://stackoverflow.com/questions/45017295/assert-an-exception-using-xunit

Reply:
For this certain, there's no message because it can only be achieve through a code error. Do u recommend I add a message still?